src/main/java/app/view/SideMenu.java

```diff
@@ -21,26 +30,137 @@ public SideMenu(int width, int height)
 
     public void loadButtons(VBox vbox)
     {
-        // Spawn areas
-        Label spawns = new Label("Spwan Areas:");
-        Button aSpawn = new Button("Agent Spawn Area");
-        Button iSpawn = new Button("Intruder Spawn Area");
-        vbox.getChildren().addAll(spawns, aSpawn, iSpawn);
+        // User Rectangle Area
+        Label l = new Label("Enter MinX, MinY, width, height:");
+        TextField minX = new TextField();
+        TextField minY = new TextField();
+        TextField width = new TextField();
+        TextField height = new TextField();
+
+        vbox.getChildren().addAll(l, minX, minY, width, height);
+
+        // Portal co-ordinates
+        Label teleport = new Label("Teleport to x and y:");
+        TextField x = new TextField();
+        TextField y = new TextField();
+
+        vbox.getChildren().addAll(teleport, x, y);
 
         // Furniture type enums
         Label furniture = new Label("Furniture Items:");
         vbox.getChildren().add(furniture);
         for(FurnitureType ft : FurnitureType.values())
         {
             Button furnType = new Button(""+ft);
+            furnType.setOnAction(e -> {
+                DrawRectangle dr = new DrawRectangle();
```

Comment:
This is one really big lambda expression, consider moving into it's own function like this:

```java
    furnType.setOnAction( e -> handleActionEvent(e));
    // other code here ...
}

private void handleActionEvent(MouseEvent e)
{
    e.getX(); // Do stuff here
}
```

Reply:
Added the handleActionEvent method.